tests/js/plugin/enhancedanchoring-test.coffee

```diff
@@ -59,178 +36,212 @@ describe 'Annotator.Plugin.EnhancedAnchoring', ->
-
-      assert.isArray anchors
-      assert.include anchors, anchor11
-      assert.include anchors, anchor12
-      assert.include anchors, anchor2
-      assert.notInclude anchors, anchor3
-
-  describe 'getHighlights', ->
-    it 'returns an empty array by default', ->
-      am = createAnchoringManager()
-      hls = am.getHighlights()
-      assert.isArray hls
-      assert.equal hls.length, 0
-
-    it 'returns all the highlights', ->
-      am = createAnchoringManager()
-      ann1 = createTestAnnotation "a1", 2
-      ann2 = createTestAnnotation "a2"
-      ann3 = createTestAnnotation "a3"
-      anchor11 = am.createAnchor(ann1, ann1.target[0]).result
-      anchor12 = am.createAnchor(ann1, ann1.target[1]).result
-      anchor2 = am.createAnchor(ann2, ann2.target).result
-      anchor3 = am.createAnchor(ann3, ann2.target).result
-
-      hls = am.getHighlights()
-
-      assert.isArray hls
-      assert.include hls, anchor11.highlight[anchor11.startPage]
-      assert.include hls, anchor12.highlight[anchor12.startPage]
-      assert.include hls, anchor2.highlight[anchor2.startPage]
-      assert.include hls, anchor3.highlight[anchor3.startPage]
-
-    it 'returns the highlights belonging to a set of annotations', ->
-      am = createAnchoringManager()
-      ann1 = createTestAnnotation "a1", 2
-      ann2 = createTestAnnotation "a2"
-      ann3 = createTestAnnotation "a3"
-      anchor11 = am.createAnchor(ann1, ann1.target[0]).result
-      anchor12 = am.createAnchor(ann1, ann1.target[1]).result
-      anchor2 = am.createAnchor(ann2, ann2.target).result
-      anchor3 = am.createAnchor(ann3, ann2.target).result
-
-      hls = am.getHighlights [ann1, ann2]
-
-      assert.isArray hls
-      assert.include hls, anchor11.highlight[anchor11.startPage]
-      assert.include hls, anchor12.highlight[anchor12.startPage]
-      assert.include hls, anchor2.highlight[anchor2.startPage]
-      assert.notInclude hls, anchor3.highlight[anchor3.startPage]
-
-  describe 'Anchor.scrollIntoView()', ->
-    it 'calls scrollIntoView() on the highlight', ->
-      am = createAnchoringManager()
-      ann = createTestAnnotation "a1"
-      anchor = am.createAnchor(ann, ann.target[0]).result
-      anchor.scrollIntoView().then ->
-
-        assert.called anchor.highlight[anchor.startPage].scrollIntoView
+        assert.ok hl
+        assert.equal hl.page, page
+
+      it 'announces the creation of the highlights in an event', ->
+        am = createAnchoringManager()
+
+        assert.notCalled am.annotator.publish
+
+        ann = createTestAnnotation "a1"
+        anchor = am.createAnchor(ann, ann.target[0]).result
+
+        hl = anchor.highlight[anchor.startPage]
+
+        assert.calledWith am.annotator.publish, 'highlightsCreated', [hl]
+
+      it 'adds an anchor property to the created Highlights', ->
+        am = createAnchoringManager()
+        ann = createTestAnnotation "a1"
+        anchor = am.createAnchor(ann, ann.target[0]).result
```

Comment:
Aron said:

> My general rule of thumb is to repeat setup in the method if its just a few lines, otherwise look at ways of moving it into a setup helper. The more contained the test the better, so **it's a matter of balance**.

Which I would agree with. Here, on balance, I'd say that repeating the same three lines eight times is probably too verbose. Something like the below would be easier to read:

```coffeescript
describe 'Annotator.Plugin.EnhancedAnchoring', ->
  ...

  describe "single-phase anchoring", ->
    am = null
    ann = null
    anchor = null

    before ->
      am = createAnchoringManager()
      ann = createTestAnnotation "a1"
      anchor = am.createAnchor(ann, ann.target[0]).result

    it 'creates the anchors from the right targets', ->
      assert.equal anchor.id, "fake anchor for " + anchor.target.id

    it '...', ->
      ...
```